libs/data-types/src/lib/utilities/bootstrap-dynamic-types/bootstrap-dynamic-types.ts

```diff
@@ -0,0 +1,202 @@
+import { isNonEmptyString, isNullOrUndefined } from '@coscrad/validation-constraints';
+import { COSCRAD_DATA_TYPE_METADATA } from '../../constants';
+import {
+    getUnionMemberMetadata,
+    getUnionMetadata,
+    isUnionMemberMetadata,
+    isUnionMetadata,
+} from '../../decorators';
+import { CoscradPropertyTypeDefinition, isSimpleCoscradPropertyTypeDefinition } from '../../types';
+import getCoscradDataSchema from './../getCoscradDataSchema';
+import { Ctor } from './../getCoscradDataSchemaFromPrototype';
+import { leveragesUniontype } from './leverages-union-type';
+import { resolveMemberSchemasForUnion } from './resolve-member-schemas-for-union';
+
+type ComplexCoscradDataTypeDefinition = {
+    complexDataType: string;
+};
+
+export type UnionTypesMap = Map<string, Map<string | number, Ctor<Object>>>;
+
+// We duplicate this here to avoid circular build dependencies
+const isComplexCoscradDataTypeDefinition = (
+    input: unknown
+): input is ComplexCoscradDataTypeDefinition =>
+    isNonEmptyString((input as ComplexCoscradDataTypeDefinition).complexDataType);
+
+export const buildUnionTypesMap = (allCtorCandidates: unknown[]): UnionTypesMap => {
+    const unionMetadata = allCtorCandidates.map(getUnionMetadata).filter(isUnionMetadata);
+
+    const unionNames = [...new Set(unionMetadata.map(({ unionName }) => unionName))];
+
+    const unionMap: UnionTypesMap = new Map();
+
+    unionNames.forEach((nameOfUnionToValidate) => {
+        const allDiscriminantPathsRegisteredForThisUnion = [
+            ...new Set(unionMetadata.map(({ discriminantPath }) => discriminantPath)),
+        ];
+
+        // TODO It would be nice to design this away
```

Comment:
Put a little thought to this before removing the comment.

Reply:
I updated the comment with a usage example. This would be good to move to documentation at some point.